internal/client/gandi.go

```diff
@@ -27,13 +24,17 @@ type Gandi struct {
 }
 
 // New instantiates a new Gandi client
-func New(apikey string, sharingID string, debug bool, dryRun bool) *Gandi {
-	return &Gandi{apikey: apikey, endpoint: gandiEndpoint, sharingID: sharingID, debug: debug, dryRun: dryRun}
+func New(apikey string, apiurl string, sharingID string, debug bool, dryRun bool) *Gandi {
```

Comment:
maybe this function can receive config.Config as its only argument, WDYT?

Reply:
I hesitated to do it, but i didn't for two reason:
- we would like to add the endpoint (instead of using the `SetEndpoint` method)
- since it is internal stuffs, we could think about this in another PR without impacting the public interface.